eureka2-ext/eureka2-eureka1x-rest-api/src/main/java/com/netflix/eureka2/eureka1x/rest/query/AbstractEureka2RegistryView.java

```diff
@@ -0,0 +1,61 @@
+package com.netflix.eureka2.eureka1x.rest.query;
+
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import com.netflix.eureka2.interests.ChangeNotification;
+import com.netflix.eureka2.interests.ChangeNotifications;
+import com.netflix.eureka2.registry.instance.InstanceInfo;
+import rx.Observable;
+import rx.Scheduler;
+import rx.Subscription;
+import rx.functions.Func1;
+import rx.subjects.BehaviorSubject;
+
+/**
+ * @author Tomasz Bak
+ */
+public abstract class AbstractEureka2RegistryView<T> {
+
+    private final Observable<ChangeNotification<InstanceInfo>> notifications;
+    private final long refreshIntervalMs;
+    private final Scheduler scheduler;
+
+    private final BehaviorSubject<T> latestCopySubject = BehaviorSubject.create();
+    private Subscription subscription;
+
+    protected AbstractEureka2RegistryView(Observable<ChangeNotification<InstanceInfo>> notifications,
+                                          long refreshIntervalMs,
+                                          Scheduler scheduler) {
+        this.notifications = notifications;
+        this.refreshIntervalMs = refreshIntervalMs;
+        this.scheduler = scheduler;
+    }
+
+    /**
+     * Connect happens immediately after child object is constructed.
+     */
+    protected void connect() {
+        subscription = notifications
```

Comment:
These all views are connected to the same local registry (which is reads server registry). We do not do any remote calls here.
We could still save some cycles by doing conversion from v1 to v2 once, but I intentionally kept it simple. We do 30 sec batching (by default), so the load from it will be minimal anyway.